### PR TITLE
(#1226) Delete Andand introduce ForEach

### PR DESCRIPTION
**Pull request by ryoku**
This is for issue (#1226)

Extracted `ForEach` from `And`
Added `ForEachTest`
Extracted `ForEachInThreads` from `AndInThreads`
Added `ForEachInThreadsTest`
Updated README.md accordingly

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1272](https://github.com/yegor256/cactoos/issues/1272) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1272](https://github.com/yegor256/cactoos/issues/1272) is assigned to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), here is [why](https://www.0crat.com/footprint/C63314D6Z/fd0a3772-d968-466a-83a6-25cfe95c0ab9); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/main/java/org/cactoos/iterator/ForEach.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.iterator;
```
Comment (fanifieiev) [inline src/main/java/org/cactoos/iterator/ForEach.java]: @ryoku The class should be located in `org.cactoos.func` package.

File: src/main/java/org/cactoos/iterator/ForEachInThreads.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.iterator;
```
Comment (fanifieiev) [inline src/main/java/org/cactoos/iterator/ForEachInThreads.java]: @ryoku The class should be located in `org.cactoos.func` package.

File: src/test/java/org/cactoos/iterator/ForEachInThreadsTest.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.iterator;
```
Comment (fanifieiev) [inline src/test/java/org/cactoos/iterator/ForEachInThreadsTest.java]: @ryoku The class should be located in `org.cactoos.func` package.

File: src/test/java/org/cactoos/iterator/ForEachTest.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.iterator;
```
Comment (fanifieiev) [inline src/test/java/org/cactoos/iterator/ForEachTest.java]: @ryoku The class should be located in `org.cactoos.func` package.

File: src/main/java/org/cactoos/iterator/ForEach.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+ *
+ * @param <X> The type to itetare over
+ * @since 0.44
+ */
+public final class ForEach<X> implements Proc<Iterable<X>> {
+
+    /**
+     * The proc.
+     */
+    private final Func<X, Boolean> wrapped;
```
Comment (fanifieiev) [inline src/main/java/org/cactoos/iterator/ForEach.java]: @ryoku Please change `wrapped` variable name to `func`

File: src/main/java/org/cactoos/iterator/ForEachInThreads.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+ *
+ * @param <X> The type to itetare over
+ * @since 0.44
+ */
+public final class ForEachInThreads<X> implements Proc<Iterable<X>> {
+
+    /**
+     * The proc.
+     */
+    private final Func<X, Boolean> wrapped;
```
Comment (fanifieiev) [inline src/main/java/org/cactoos/iterator/ForEachInThreads.java]: @ryoku Please change `wrapped` variable name to `func`

**Comment by fanifieiev** [comment]
@paulodamaso Please merge

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Code review was too long (20 days), architects (@paulodamaso) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by paulodamaso** [comment]
@rultor release tag is `0.44`

**Comment by paulodamaso** [comment]
@rultor release tag is `0.44`

**Pull request merged**